src/femr/labelers/omop.py

```diff
@@ -463,178 +594,151 @@ def get_labeler_type(self) -> LabelType:
-    coded occurrence(s) of Anemia in `time_horizon`."""
-
-    # fmt: off
-    original_omop_concept_codes = [
-        'SNOMED/271737000', 'SNOMED/713496008', 'SNOMED/713349004', 'SNOMED/767657005',
-        'SNOMED/111570005', 'SNOMED/691401000119104', 'SNOMED/691411000119101',
-    ]
-    # fmt: on
+    patients_to_labels: Dict[int, List[Label]] = {}
+    for patient_id in patient_ids:
+        patient: Patient = patients[patient_id]  # type: ignore
+        patient_df = chexpert_df[chexpert_df["patient_id"] == patient_id]
 
+        if num_labels is not None and num_labels < len(patient_df):
+            patient_df = patient_df.sample(n=num_labels, random_state=0)
+        labels: List[Label] = labeling_function.label(patient, patient_df)
+        patients_to_labels[patient_id] = labels
 
-class HyperkalemiaCodeLabeler(OMOPConceptCodeLabeler):
-    """Apply a label for whether a patient has at 1+ explicitly
-    coded occurrence(s) of Hyperkalemia in `time_horizon`."""
+    return patients_to_labels
 
-    # fmt: off
-    original_omop_concept_codes = [
-        'SNOMED/14140009',
-    ]
-    # fmt: on
 
+class ChexpertLabeler(Labeler):
+    """CheXpert labeler.
 
-class HyponatremiaCodeLabeler(OMOPConceptCodeLabeler):
-    """Apply a label for whether a patient has at 1+ explicitly
-    coded occurrence(s) of Hyponatremia in `time_horizon`."""
+    Multi-label classification task of patient's radiology reports.
+    Make prediction 24 hours before radiology note is recorded.
 
-    # fmt: off
-    original_omop_concept_codes = [
-        'SNOMED/267447008', 'SNOMED/89627008'
-    ]
-    # fmt: on
+    Excludes:
+        - Radiology reports that are written <=24 hours of a patient's first event (i.e. `patient.events[0].start`)
+    """
 
+    def __init__(
+        self,
+        path_to_chexpert_csv: str,
+    ):
+        self.path_to_chexpert_csv = path_to_chexpert_csv
 
-class ThrombocytopeniaCodeLabeler(OMOPConceptCodeLabeler):
-    """Apply a label for whether a patient has at 1+ explicitly
-    coded occurrence(s) of Thrombocytopenia in `time_horizon`."""
+    def get_patient_start_end_times(self, patient: Patient) -> Tuple[datetime.datetime, datetime.datetime]:
+        """Return the (start, end) of the patient timeline.
 
-    # fmt: off
-    original_omop_concept_codes = [
-        'SNOMED/267447008', 'SNOMED/89627008',
-    ]
-    # fmt: on
+        Returns:
+            Tuple[datetime.datetime, datetime.datetime]: (start, end)
+        """
+        return (patient.events[0].start, patient.events[-1].start)
 
+    def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
+        """Return a list of all times when the patient has a radiology report"""
 
-class NeutropeniaCodeLabeler(OMOPConceptCodeLabeler):
-    """Apply a label for whether a patient has at 1+ explicitly
-    coded occurrence(s) of Neutkropenia in `time_horizon`."""
+        chexpert_df = pd.read_csv(self.path_to_chexpert_csv, sep="\t")
```

Comment:
I really would not use Pandas here. Pandas's date handling in particular is terrible, which has caused no end of headaches for the PE project.

Reply:
I think this is fine for this one use case since we know the exact dataset and that the dates work out.